### ISSUE TITLE
@auth multi-owner rule: undefined ownerField on record grants access incorrectly

### ISSUE THREAD

**Issue by lennybr** [aws-amplify/amplify-cli#344]
**Describe the bug**
After #285 when using a multi-owner auth rule, if the value of the ownerField on a record is undefined rather than an empty list, the authorisation rule grants access by default instead of filtering it out for all users.

**To Reproduce**
1. Add to your schema
```
type Draft
    @auth(rules: [
        {allow: owner},
        {allow: owner, ownerField: "editors"}
    ])
    @model
{
    id: ID!
    owner: String
    editors: [String]
}
```
2. In **Query.listDraft.response** the authorisation code for a multi-owner field is currently generated as
```
...
## Authorization rule: { allow: "owner", ownerField: "editors", identityField: "username" } **
#set( $allowedOwners1 = $item.editors )
#set( $identityValue = $util.defaultIfNull($ctx.identity.username, "___xamznone____") )
#if( $util.isList($allowedOwners1) )
  #foreach( $allowedOwner in $allowedOwners1 )
    #if( $allowedOwner == $identityValue )
      #set( $isLocalOwnerAuthorized = true )
    #end
  #end
#end
...
```
3. Create a new Draft record
```
mutation CreateDraft {
  createDraft(input: {}) {
    id
    owner
    editors
  }
}
```
4. The new draft record's **editors** value is undefined (can also accomplish this by unsetting the editors value).
5. Authenticate as another user (or change the owner value to simulate).
6. The query below returns the draft record even though the user is no longer authorised to access this record by the {allow: owner} rule, and in theory if there are no editors, then the record should **not** be returned.
```
query listDrafts {
  listDrafts {
    items {
      id
      owner
      editors
    }
  }
}
```

**Expected behavior**
A record where a multi-owner field is not set should behave the same as if the field is empty and this record should be filtered out as part of a query response.

**Additional context**
It looks like this is being caused by the authorisation code in **Query.listDraft.response** for multi-owner fields. I'm not sure if this is the right solution, but by changing the following generated line from
``#set( $allowedOwners1 = $item.members )``
to
``#set( $allowedOwners1 = $util.defaultIfNull($item.members, []))``
this filtered correctly.

It likely affects other queries and mutations, but I haven't had a chance to dig into those yet.

**Comment by mikeparisstuff** [comment]
Please see the section titled "Ownership with create mutations" here https://aws-amplify.github.io/docs/js/graphql. In order to make owner auth as simple as possible, if you do not provide a value (aka undefined) when creating an object, the currently logged in user is automatically added as the owner. If the owner is a String then the user is the sole owner at that field and if the owner is a list of strings the user is inserted as single element in the list.

If you want to create an item that does not automatically fill the list run the mutation like this:

```
mutation CreateDraft {
  createDraft(input: { owner: null, editors: [] }) {
    id
    owner
    editors
  }
}
```

With this being said, I think your comment about `#set( $allowedOwners1 = $util.defaultIfNull($item.members, []))` is valid and I can make sure it is explicit about the default when the field is undefined or null.

**Comment by lennybr** [comment]
Thanks @mikeparisstuff. The explicit deny access when the ownerField is missing seems more "secure" by default.

**Comment by mikeparisstuff** [comment]
@lennybr That is fair. The inspiration for adding automatically was to prevent you from having to dig the value for your identityField out of the Cognito user on the client every time while still allowing you to be explicit when you wanted. I'm curious what other people think about this behaviour. Please feel free to chime in.

**Comment by mikeparisstuff** [comment]
Closing as this is addressed by #1043.
